wikis.go

```diff
@@ -0,0 +1,184 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// WikisService handles communication with the wikis related methods of
+// the Gitlab API Introduced in GitLab 10.0.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html
+type WikisService struct {
+	client *Client
+}
+
+// WikiFormat represents the available formats that can be used for a wiki page
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html
+type WikiFormat string
+
+// The available todo actions.
+const (
+	WikiFormatMarkdown WikiFormat = "markdown"
+	WikiFormatRFoc     WikiFormat = "rdoc"
+	WikiFormatASCIIDoc WikiFormat = "asciidoc"
+)
+
+// Wiki represents a GitLab wiki.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html
+type Wiki struct {
+	Content string     `json:"content"`
+	Format  WikiFormat `json:"format"`
+	Slug    string     `json:"slug"`
+	Title   string     `json:"title"`
+}
+
+func (t Wiki) String() string {
+	return Stringify(t)
+}
+
+// ListWikisOptions represents the available ListWikis() options.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html#get-a-list-of-wikis
+type ListWikisOptions struct {
+	WithContent *bool `url:"with_content,omitempty" json:"with_content,omitempty"`
+}
+
+// ListWikis lists all pages of the wiki of the given project id.
+// When with_content is set, it returns also the content of the pages.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html#list-wiki-pages
+func (s *WikisService) ListWikis(pid interface{}, opt *ListWikisOptions, options ...OptionFunc) ([]*Wiki, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	u := fmt.Sprintf("projects/%s/wikis", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var t []*Wiki
+	resp, err := s.client.Do(req, &t)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return t, resp, err
+}
+
+// GetWikiPage gets a wiki page for a given project.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html#get-a-wiki-page
+func (s *WikisService) GetWikiPage(pid interface{}, slug string, options ...OptionFunc) (*Wiki, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	u := fmt.Sprintf("projects/%s/wikis/%s", url.QueryEscape(project), url.QueryEscape(slug))
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var t *Wiki
+	resp, err := s.client.Do(req, &t)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return t, resp, err
+}
+
+// CreateNewWikiPageOptions represents options to CreateNewWikiPage
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/wikis.html#create-a-new-wiki-page
+type CreateNewWikiPageOptions struct {
```

Comment:
IMO it's just a bit cleaner to call this type `CreateWikiPageOptions`. `Create` and `New` seem to indicate the same thing.

Reply:
Ok